templates/node/node.py

```diff
@@ -0,0 +1,277 @@
+
+    @property
+    def machine(self):
+        if self._machine:
+            return self._machine
+        return self.space.machines.get(self.name)
+
+    def install(self):
+        try:
+            self.state.check('actions', 'install', 'ok')
+            return
+        except StateCheckError:
+            pass        
+
+        machine = self.machine
+        if not machine:
+            machine = self._machine_create()
+
+        # Get data from the vm
+        ip_private, vm_info = machine.machineip_get()
+        self.data['sshLogin'] = vm_info['accounts'][0]['login']
+        self.data['sshPassword']= vm_info['accounts'][0]['password']
+        self.data['ipPrivate'] = ip_private
+        self.data['ipPublic'] = machine.space.model['publicipaddress']
+        self.data['machineId'] = machine.id
+
+        self.portforward_create(self.data['ports'])
+        self._configure_disks()
+        self.save()
+
+        self.state.set('actions', 'install', 'ok')
+
+    def _machine_create(self):
+        """ Create a new machine """
+        data = self.data
+        space = self.space
+        self._machine = space.machine_create(
+            name=self.name,
+            sshkeyname= self.sshkey,
+            image=data['osImage'],
+            disksize=data['bootDiskSize'],
+            datadisks=[data['dataDiskSize']],
+            sizeId=data['sizeId'],
+            )
+
+        return self._machine
+
+    def _configure_disks(self):
+        """
+        Configure one boot disk and one data disk when installing a machine.
+        """
+        machine = self.machine
+        space_name = machine.space.model['name']
+        if machine:
+            machine.start()
+        else:
+            raise RuntimeError('machine %s in not found' % self.name)        
+
+        for disk in machine.disks:
+            # create a disk service
+            service = self.api.services.create(
+                template_uid=self.DISK_TEMPLATE, 
+                service_name= 'Disk%s' % str(disk['id']),
+                data={'vdc' : space_name},
+            )
+            # update data in the disk service
+            task = service.schedule_action('update_data', {'data':disk})
+            task.wait()
+            # limmit disk with default limits
+            task = service.schedule_action('limit_io')
+            task.wait()                   
+
+
+    def uninstall(self):
+        """ Uninstall machine """
+        if not self.machine:
+            raise RuntimeError('machine %s in not found' % self.name)
+        self.machine.delete()
+
+    def portforward_create(self, ports):
+        """ Add portforwards """
+        if not self.machine:
+            raise RuntimeError('machine %s in not found' % self.name)
+
+        # get vdc service
+        task = self.vdc.schedule_action('portforward_create', 
+                                        {'machineId':self.machine.id, 'port_forwards':ports, 'protocol':'tcp'})
+        task.wait()
```

Comment:
Do we need to wait? i guess u only need to wait if u gonna use the results of this action. Hence waiting will just cause an over head (pulling for the action result).

Same for the portforward_delete